uds.js

```diff
@@ -295,5 +297,62 @@
         return executeUdsAjaxCall(onSuccess,onFailure,url,'GET');
     };
 
+    uds.removeUserSbr = function (onSuccess, onFailure,userId,query) {
+        if (!$.isFunction(onSuccess)) { throw 'onSuccess callback must be a function'; }
+        if (!$.isFunction(onFailure)) { throw 'onFailure callback must be a function'; }
+        var url = udsHostName.clone().setPath('/user/'+userId+'/sbr').addQueryParam('where', query);
+        return executeUdsAjaxCall(onSuccess,onFailure,url,'DELETE');
+    };
+
+
+    uds.getRoleList = function (onSuccess, onFailure, query) {
+        if (!$.isFunction(onSuccess)) { throw 'onSuccess callback must be a function'; }
+        if (!$.isFunction(onFailure)) { throw 'onFailure callback must be a function'; }
+        var url = udsHostName.clone().setPath('/user/metadata/roles');
+        url.addQueryParam('where',encodeURIComponent(query));
+        return executeUdsAjaxCall(onSuccess,onFailure,url,'GET');
+    };
+
+    uds.getRoleDetails = function (onSuccess, onFailure,parentRoleId,roleName) {
+        if (!$.isFunction(onSuccess)) { throw 'onSuccess callback must be a function'; }
+        if (!$.isFunction(onFailure)) { throw 'onFailure callback must be a function'; }
+        var url = udsHostName.clone().setPath('/user?resourceProjection=Full&where=((parentRoleId is '+parentRoleId+') or (roleName is"'+roleName+'" ))');
+        return executeUdsAjaxCall(onSuccess,onFailure,url,'GET');
+    };
+
+    uds.getRoleParentDetails = function (onSuccess, onFailure,parentUserId,parentRoleId,roleName) {
+        if (!$.isFunction(onSuccess)) { throw 'onSuccess callback must be a function'; }
+        if (!$.isFunction(onFailure)) { throw 'onFailure callback must be a function'; }
+        var url = udsHostName.clone().setPath('/user?resourceProjection=Full&where=((parentRoleId is '+parentRoleId+' and roleParentUserId is "'+parentUserId+'") or (roleName is"'+roleName+'" and roleUserId is "'+parentUserId+'"))');
```

Comment:
Just suggestion, can we use addQueryParam for where, so that we will have consistent coding style.